File: command-line_script_template.pl
#!/usr/bin/perl

# this script is based on examples from
# http://perldoc.perl.org/Getopt/Long.html#SYNOPSIS
#
# Pod::Usage can be used for more universal knobs parsing, but seems to be
# not working in old perl versiond
#
# Getopt::Long::Descriptive can be used instead of Getopt::Long + Pod::Usage,
# but is not included in Perl by default

use Getopt::Long;

my $help_message = "Usage: $0 [options]\n
Script description\n
Specific options:\n
	-f, --file=FILE		Set file path to FILE
\nCommon options:\n
	-h, --help		Show this message
";
my $file;
GetOptions (
        "file=s" => \$file, # -file will set string value to $file
        "help", # -help will set $opt_help to true
        ) or die($help_message);     
print $help_message if $opt_help;

# script code

File: command-line_script_template.rb
#!/usr/bin/ruby

# this script is base on example from
# http://ruby-doc.org/stdlib-2.2.0/libdoc/optparse/rdoc/OptionParser.html

require 'optparse'

options = {}
OptionParser.new do |opts|
  opts.banner = "Usage: #{File.basename(__FILE__)} [options]\n\n"
  opts.define_head 'Script description'

  opts.separator ''
  opts.separator 'Specific options:'

  opts.on('-fFile', '--file=FILE', 'Set file path to FILE') do |f|
    options[:file_path] = f
  end

  opts.separator ''
  opts.separator 'Common options:'

  opts.on_tail('-h', '--help', 'Show this message') do
    puts opts
    exit
  end
end.parse!

# script code (using options hash)
